src/main/java/flaxbeard/immersivepetroleum/common/blocks/metal/TileEntityPumpjack.java

```diff
@@ -146,6 +147,31 @@ public void update()
 		update(true);
 	}
 
+	// Checks whether or not this pump has a pipe to bedrock.
+	private boolean hasPipe() {
+		if (!IPConfig.Extraction.require_pipes) {
+			return true;
+		}
+		
+		World world = this.getWorld();
+		BlockPos pos = this.getOrigin();
+		int depthCount = 0;
+		int depthBounds = 256;
```

Comment:
you should change this to only go down from pos.getY() rather than the entire world all the time